src/main/scala/subsystem/HasTiles.scala

```diff
@@ -86,6 +91,49 @@ trait HasTileInterruptSources
   }
 }
 
+/** These are sources of "constants" that are driven into the tile.
+  * While they are not expected to change dyanmically while the tile is executing code,
+  * they may be either tied to a contant value or programmed during boot or reset.
+  * They need to be instantiated before tiles are attached to the subsystem containing them.
+  */
+trait HasTileInputConstants extends InstantiatesTiles { this: BaseSubsystem =>
+  // Collect dynamic hart id prefixes and/or distribute static hart ids to the tiles
+  val tileHartIdNode = BundleBridgeEphemeralNode[UInt]()
+  val tileHartIdNexusNode = BundleBridgeNexus[UInt](
+    inputFn = BundleBridgeNexus.orReduction[UInt](p(HartPrefixKey)) _,
+    outputFn = (prefix: UInt, n: Int) =>  Seq.tabulate(n) { i =>
+      val y = dontTouch(prefix | hartIdList(i).U(p(MaxHartIdBits).W))
+      if (p(HartPrefixKey)) BundleBridgeNexus.safeRegNext(y) else y
+    },
+    default = Some(() => 0.U(p(MaxHartIdBits).W))
+  )
+  // TODO: this also needs to be wired into the DebugModule's hart selection circuit?
```

Comment:
The debug module as currently written requires in its parameters a function UInt <-> UInt to map hartID (as seen by the core in its `mhartid` CSR) to hart index. 

However in the special case that you only have one core, the debug module is able to ignore dynamic hart ID's. Otherwise it doesn't work with dynamic hart IDs. Not sure if you are able to solve this in this PR? If so add @ernie-sifive as a reviewer...

Reply:
Yeah I think dealing with this kind of boot-time dynamism is something that is going to have to be solved as part of a certain internal project, but not this PR.